test.py

```diff
@@ -262,8 +262,10 @@ def boost_tests(self):
         return []
 
     def build_test_list(self) -> List[str]:
-        return [os.path.splitext(t.relative_to(self.suite_path))[0] for t in
-                self.suite_path.glob(self.pattern)]
+        path = self.suite_path
+        # Search the test dir recursively
+        tests = path.rglob(self.pattern)
+        return [os.path.splitext(t.relative_to(self.suite_path))[0] for t in tests]
```

Comment:
I remember (and now double-checked I am right) that PythonTestSuite has its own implementation of this function, which already used rglob. So why is this change needed - only for non-Python tests? Or, if you want to do it in the parent class, maybe we can remove it from the child class?

Reply:
@temichus this patch is marked as fixing the issue you created, so please review this fix.